matching-notification-ui/src/main/resources/MatchingNotification/MatchesTable.xml

```diff
@@ -127,20 +129,44 @@ $xwiki.jsx.use('PhenoTips.Widgets')##
+                    #set($endTime = $matchesUpdateObject.getValue('completedTime'))##
+                  #end##
+                  &lt;input type="hidden" class="disclaimer" value="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.disclaimer.PhenomeCentral'))"&gt;##
+                  &lt;input type="hidden" class="endTime" value="${escapetool.xml($!endTime)}"&gt;##
+                  &lt;input type="hidden" class="serverName" value="$escapetool.xml($services.localization.render('phenotips.matchingNotifications.table.matchesSeletFilter.local'))"&gt;##
+                  &lt;span class="fa fa-info-circle mme-disclaimer-help" title="$escapetool.xml($services.localization.render('phenotips.myMatches.disclaimer.showDisclaimer'))"&gt;&lt;/span&gt;
+                  &lt;br&gt;
                 &lt;/label&gt;
                 &lt;div class="across-mme"&gt;&lt;input id="mme-filter" type="checkbox" checked &gt; $escapetool.xml($services.localization.render('phenotips.myMatches.filters.mme.label')) &lt;a href="http:\/\/matchmakerexchange.org" target="_blank"&gt;MME&lt;/a&gt;:&lt;/div&gt;
                 #set ($preferenceDoc = $xwiki.getDocument("XWiki.XWikiPreferences"))
                 #set ($allRemotes = $preferenceDoc.getObjects("PhenomeCentral.RemoteMatchingServiceConfiguration"))
                 #foreach ($config in $allRemotes)
                   #if ($config.getProperty('serverId').getValue() != "" &amp;&amp; $config.getProperty('humanReadableName').getValue() != "" &amp;&amp; $config.getValue('includeInUI') == 1)
-                    &lt;label&gt;&lt;input name="checkbox-server-id-filter" class="mme" type="checkbox" value="$config.getProperty('serverId').getValue()" checked&gt;
-                      &lt;div class="filter-mme-server-name"&gt;$escapetool.xml($config.getProperty('humanReadableName').getValue())&lt;span class="matches-count"&gt;&lt;/span&gt;&lt;/div&gt;&lt;br&gt;
+                    #set ($serverID = $config.getProperty('serverId').getValue())
+                    #set ($serverName = $config.getProperty('humanReadableName').getValue())
+                    #if ($serverName == "")
+                      #set ($serverName = $serverID)
+                    #end
+                    #set ($disclaimer = $config.getProperty('disclaimer').getValue())
+                    #set ($matchesUpdateObject = $matchesRunDoc.getObject($infoClassname, 'serverName', $serverID))
+                    #if ($matchesUpdateObject)
+                      #set($endTime = $matchesUpdateObject.getValue('completedTime'))
+                    #end
+                    &lt;label&gt;
+                      &lt;input name="checkbox-server-id-filter" class="mme" type="checkbox" value="${escapetool.xml($serverID)}" checked&gt;
+                      &lt;input type="hidden" class="disclaimer" value="${escapetool.xml($disclaimer)}"&gt;
+                      &lt;input type="hidden" class="endTime" value="${escapetool.xml($!endTime)}"&gt;
```

Comment:
Please see the screenshots below. Even though I never ran matches on the remote server called "Not matching", the a non-empty endTime is reported at it's identical to the one recorded the remote server above it, named "Local (self)".

 ![image](https://user-images.githubusercontent.com/651980/63042123-61b37200-be97-11e9-9ace-29530d156630.png)

![image](https://user-images.githubusercontent.com/651980/63042190-798af600-be97-11e9-8c60-01632995204c.png)